src/merkle_tree.ml

```diff
@@ -408,6 +408,17 @@ struct
     let typ ~depth : (var, value) Typ.t = Typ.(list ~length:depth Hash.typ)
   end
 
+  module Tag = struct
+    type value = [`Curr_ledger | `Epoch_ledger]
+
+    type var = Boolean.var
+
+    let typ : (var, value) Typ.t =
```

Comment:
I think this would amount to the same using the [`Enumerable`](https://github.com/o1-labs/snarky/blob/master/src/enumerable.mli) interface, might be cleaner to do it that way.